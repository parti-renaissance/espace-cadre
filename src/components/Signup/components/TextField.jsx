import { TextField as MuiTextField } from '@mui/material'
import { styled } from '@mui/system'
import PropTypes from 'prop-types'
import AlertBanner from '~/ui/AlertBanner'

const TextInput = styled(MuiTextField)(
  ({ theme }) => `
  border-radius: 8px;
  outline: none;
  color: ${theme.palette.blackCorner};
  background-color: ${theme.palette.gray100};
  
  & .MuiOutlinedInput-notchedOutline {
    border: none;
  }
`
)

export const TextFieldFormik = ({ formik, label, inputProps = { maxLength: 255 }, ...props }) => (
  <>
    <TextInput
      {...props}
      size="medium"
      error={!!formik.touched[label] && !!formik.errors[label]}
      fullWidth
      id={label}
      name={label}
      inputProps={inputProps}
      value={formik.values[label]}
      onChange={formik.handleChange}
    />
    {formik.touched[label] && formik.errors[label] && <AlertBanner severity="error" message={formik.errors[label]} />}
  </>
)

TextFieldFormik.propTypes = {
  formik: PropTypes.object.isRequired,
  label: PropTypes.string.isRequired,
  inputProps: PropTypes.object,
}

export const TextField = ({ value = '', inputProps = { maxLength: 255 }, onChange = () => {}, ...props }) => (
  <TextInput {...props} size="medium" inputProps={inputProps} value={value} onChange={onChange} />
)

TextField.propTypes = {
  value: PropTypes.string,
  onChange: PropTypes.func,
  inputProps: PropTypes.object,
}
