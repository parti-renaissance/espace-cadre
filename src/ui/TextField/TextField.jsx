import { TextField as MuiTextField } from '@mui/material'
import { styled } from '@mui/system'
import PropTypes from 'prop-types'
import AlertBanner from '~/ui/AlertBanner'

const TextInput = styled(MuiTextField)(
  ({ theme }) => `
  border-radius: 8px;
  margin: ${theme.spacing(1, 0)};

  & .MuiOutlinedInput-root {
    border-radius: 8px;
  }
`
)

const TextField = ({ formik, label, inputProps = { maxLength: 255 }, ...props }) => (
  <>
    <TextInput
      {...props}
      error={!!formik.touched[label] && !!formik.errors[label]}
      fullWidth
      size="medium"
      variant="outlined"
      multiline={label === 'body'}
      id={label}
      name={label}
      inputProps={inputProps}
      value={formik.values[label]}
      onChange={formik.handleChange}
    />
    {formik.touched[label] && formik.errors[label] && <AlertBanner severity="error" message={formik.errors[label]} />}
  </>
)

export default TextField

TextField.propTypes = {
  formik: PropTypes.object.isRequired,
  label: PropTypes.string.isRequired,
  inputProps: PropTypes.object,
}
