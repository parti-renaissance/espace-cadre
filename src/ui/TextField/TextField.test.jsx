import TextField from './TextField'
import { render } from '@testing-library/react'

vi.mock('@mui/material', () => ({
  TextField: ({ fullWidth, multiline, inputProps, error, ...props }) => (
    <div className="mui-TextField-mock" {...props} />
  ),
}))
vi.mock('ui/AlertBanner', () => ({
  default: ({ children, ...rest }) => (
    <div className="AlertBannerMock" {...rest}>
      {children}
    </div>
  ),
}))

describe('TextField', () => {
  const formik = {
    touched: {
      foo: 'foo-touched',
    },
    errors: {
      foo: 'foo-errors',
    },
    values: {
      foo: 'foo-values',
    },
  }

  it('displays TextField', () => {
    const { container } = render(<TextField label="foo" formik={formik} />)

    expect(container).toMatchSnapshot()
  })
})
