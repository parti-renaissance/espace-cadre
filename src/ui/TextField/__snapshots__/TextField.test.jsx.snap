// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TextField > displays TextField 1`] = `
<div>
  <div
    class="css-1yf0ffy"
    id="foo"
    name="foo"
    value="foo-values"
    variant="outlined"
  />
  <div
    class="AlertBannerMock"
    message="foo-errors"
    severity="error"
  />
</div>
`;

exports[`TextField displays TextField 1`] = `
<div>
  <div
    class="css-1yf0ffy"
    id="foo"
    name="foo"
    value="foo-values"
    variant="outlined"
  />
  <div
    class="AlertBannerMock"
    message="foo-errors"
    severity="error"
  />
</div>
`;
